drivers/SmartThings/zigbee-button/src/zigbee-multi-button/robb/init.lua

```diff
@@ -0,0 +1,191 @@
+    offset = 0xF0
+    pressed_type = "down_hold"
+  end
+
+  local event = capabilities.button.button[pressed_type](additional_fields)
+  device:emit_event_for_endpoint(ep + offset, event)
+  device:emit_event(event)
+end
+
+local do_configuration = function(driver, device)
+  -- Get the right number of endpoints
+  local endpoints = WIRELESS_REMOTE_FINGERPRINTS[device:get_model()].endpoints
+
+  device:send(device_management.build_bind_request(device, PowerConfiguration.ID, driver.environment_info.hub_zigbee_eui), 1)
+  device:send(PowerConfiguration.attributes.BatteryPercentageRemaining:configure_reporting(device, 30, 21600, 1))
+
+  for endpoint = 1, endpoints do
+    device:send(device_management.build_bind_request(device, Level.ID, driver.environment_info.hub_zigbee_eui, endpoint))
+    device:send(device_management.build_bind_request(device, OnOff.ID, driver.environment_info.hub_zigbee_eui, endpoint))
+  end
+
+  device:send(OnOff.attributes.OnOff:configure_reporting(device, 0, 600, 1))
+
+  device:send(Basic.attributes.DeviceEnabled:write(device, true))
+  if not device:get_field('is_group_configured') then
+    -- Configure adding hub to group once
+    driver:add_hub_to_zigbee_group(0xE902)
+    -- Add two more groups if it is the 8x wall switch
+    if endpoints == 4 then
+      driver:add_hub_to_zigbee_group(0xE903)
+      driver:add_hub_to_zigbee_group(0xE904)
+    end
+
+    device:set_field('is_group_configured', true, { persist = true })
+  end
+end
+
+local function added_handler(self, device)
+  local supported_button_values
+  local number_of_buttons
+  local comp_id
+
+  for _, comp in pairs(device.profile.components) do
+    comp_id = comp.id
+
+    -- Special handeling of ON buttons (left column)
+    if comp_id == "button1" or comp_id == "button3" or comp_id == "button5" or comp_id == "button7" then
+      supported_button_values = { "pushed", "up_hold" }
+      number_of_buttons = 1
+    -- Special handling of OFF buttons (right column)
+    elseif comp_id == "button2" or comp_id == "button4" or comp_id == "button6" or comp_id == "button8" then
+      supported_button_values = { "pushed", "down_hold" }
+      number_of_buttons = 1
+    -- Handling of main button capability
+    elseif comp_id == "main" then
+      supported_button_values = { "pushed", "up_hold", "down_hold" }
+      number_of_buttons = WIRELESS_REMOTE_FINGERPRINTS[device:get_model()].buttons
+    end
+
+    device:emit_component_event(comp,
+      capabilities.button.supportedButtonValues(supported_button_values, { visibility = { displayed = false } }))
+    device:emit_component_event(comp,
+      capabilities.button.numberOfButtons({ value = number_of_buttons }, { visibility = { displayed = false } }))
+  end
+  device:emit_event(capabilities.button.button.pushed({ state_change = false }))
+  device:send(PowerConfiguration.attributes.BatteryPercentageRemaining:read(device))
+end
+
+local battery_perc_attr_handler = function(driver, device, value, zb_rx)
```

Comment:
I have now tested both devices again with a battery dummy, where I can set the voltage.
I am really sure that the device does not follow the Zigbee specification and instead reports whole percentages.
If I implement it like this "value.value / 2.0 + 0.5" it starts with 50% at the maximum battery voltage of 3V.

The capability handler is a workaround to show battery capacity only in the main component. For the 4-button remote, the battery level is reported via EP2, for the 8-button remote via EP4. The default battery capability doesn't seem to handle this correctly and expects a capability in one of the other components 3 or 7, but UX technically it doesn't make sense there.

```2023-05-03T05:55:59.373744354+00:00 INFO Zigbee Button  <ZigbeeDevice: 557b9977-fcd4-4883-8578-7b0d7c7fc8d9 [0xF93B] (ROBB 8x Wireless Smarrt Switch)> received Zigbee message: < ZigbeeMessageRx || type: 0x02, < AddressHeader || src_addr: 0xF93B, src_endpoint: 0x04, dest_addr: 0x0000, dest_endpoint: 0xFF, profile: 0x0104, cluster: PowerConfiguration >, lqi: 0xC6, rssi: -72, body_length: 0x0007, < ZCLMessageBody || < ZCLHeader || frame_ctrl: 0x08, seqno: 0x7B, ZCLCommandId: 0x0A >, < ReportAttribute || < AttributeRecord || AttributeId: 0x0021, DataType: Uint8, BatteryPercentageRemaining: 0x64 > > > >
2023-05-03T05:55:59.393409021+00:00 TRACE Zigbee Button  Found ZigbeeMessageDispatcher handler in zigbee_button
2023-05-03T05:55:59.396347687+00:00 INFO Zigbee Button  Executing ZclClusterAttributeValueHandler: cluster: PowerConfiguration, attribute: BatteryPercentageRemaining
2023-05-03T05:55:59.404658354+00:00 WARN Zigbee Button  Attempted to generate event for 557b9977-fcd4-4883-8578-7b0d7c7fc8d9.button7 but it does not support capability Battery
```

Reply:
@zwave-mke based on the results from the cert testing, it seems like the device is sending `200 (0xC8)` when reporting 100% battery. This is what we would expect from a device that follows the zigbee spec, so it seems this device is following the zigbee spec. Based on the testing, using the `capabilities.battery.battery(math.floor(value.value / 2.0 + 0.5))` equation should fix the reporting to be 0-100%. 

You are correct that you need to override the default handler because of the unique endpoint mapping on this device. So, you should still have the `battery_perc_attr_handler`, but it should look like this with the new equation that divides the value by 2:

```
local battery_perc_attr_handler = function(driver, device, value, zb_rx)
  device:emit_event(capabilities.battery.battery(math.floor(value.value / 2.0 + 0.5)))
end
```

> If I implement it like this "value.value / 2.0 + 0.5" it starts with 50% at the maximum battery voltage of 3V.
I am wondering if there might have been an issue with the test setup? Based on the results from the cert testing, the device was reporting 200% when the battery was at 100%. Please let me know if you have further questions, but based on the cert test results we believe that the suggested fix above should address the battery reporting issue